fft/src/KokkosFFT_ROCM_types.hpp

```diff
@@ -17,27 +25,227 @@ static_assert(sizeof(std::complex<double>) == sizeof(Kokkos::complex<double>));
+
+    // Compute dist and strides from extents
+    int idist = std::accumulate(in_extents.begin(), in_extents.end(), 1,
+                                std::multiplies<>());
+    int odist = std::accumulate(out_extents.begin(), out_extents.end(), 1,
+                                std::multiplies<>());
+
+    auto in_strides  = compute_strides<int, std::size_t>(in_extents);
+    auto out_strides = compute_strides<int, std::size_t>(out_extents);
+    auto reversed_fft_extents =
+        convert_int_type_and_reverse<int, std::size_t>(fft_extents);
+
+    // Create a plan description
+    ScopedRocfftPlanDescription scoped_description;
+    rocfft_status status = rocfft_plan_description_set_data_layout(
+        scoped_description.description(),  // description handle
+        in_array_type,                     // input array type
+        out_array_type,                    // output array type
+        nullptr,                           // offsets to start of input data
+        nullptr,                           // offsets to start of output data
+        in_strides.size(),                 // input stride length
+        in_strides.data(),                 // input stride data
+        idist,                             // input batch distance
+        out_strides.size(),                // output stride length
+        out_strides.data(),                // output stride data
+        odist);                            // output batch distance
+
+    KOKKOSFFT_THROW_IF(status != rocfft_status_success,
+                       "rocfft_plan_description_set_data_layout failed");
+
+    // inplace or Out-of-place transform
+    const rocfft_result_placement place =
+        is_inplace ? rocfft_placement_inplace : rocfft_placement_notinplace;
+
+    // Create a plan
+    status = rocfft_plan_create(&m_plan, place, fft_direction, m_precision,
+                                reversed_fft_extents.size(),  // Dimension
+                                reversed_fft_extents.data(),  // Lengths
+                                howmany,  // Number of transforms
+                                scoped_description.description()  // Description
+    );
+    KOKKOSFFT_THROW_IF(status != rocfft_status_success,
+                       "rocfft_plan_create failed");
+
+    m_is_plan_created = true;
+
+    // Prepare workbuffer and set execution information
+    status = rocfft_execution_info_create(&m_execution_info);
+    KOKKOSFFT_THROW_IF(status != rocfft_status_success,
+                       "rocfft_execution_info_create failed");
+
+    m_is_info_created = true;
+
+    // set stream
+    // NOTE: The stream must be of type hipStream_t.
+    // It is an error to pass the address of a hipStream_t object.
+    hipStream_t stream = exec_space.hip_stream();
+    status = rocfft_execution_info_set_stream(m_execution_info, stream);
+    KOKKOSFFT_THROW_IF(status != rocfft_status_success,
+                       "rocfft_execution_info_set_stream failed");
```

Comment:
I would also wrap `rocfft_execution_info`. If `rocfft_execution_info_set_stream` fails, `m_execution_info` will never be freed.

Reply:
That is true. I will do that